### PR TITLE
Implement day_of_week for BQ

### LINKED ISSUE #1494: Implement day_of_week for BigQuery
Extend support for extraction logic to BQ where currently lacking.

Here's some code: https://github.com/ibis-project/ibis/blob/master/ibis/expr/types.py#L735-L743

### PR DESCRIPTION
**Pull request by missing-semicolon**
Closes #1494

### CODE REVIEW

**Comment by missing-semicolon** [comment]
Strange linting here...

File: ibis/bigquery/tests/test_client.py [hunk 1]
```diff
@@ -276,13 +275,15 @@ def test_scalar_param_boolean(alltypes, df):
     ).sort_values('id').reset_index(drop=True)
     expected = df.loc[
         df.string_col.astype('int64') == 0
-    ].sort_values('id').reset_index(drop=True)
+        ].sort_values('id').reset_index(drop=True)
```
Comment (cpcloud) [inline ibis/bigquery/tests/test_client.py]: You shouldn't need to make this change. The lint errors are only on a few lines: https://circleci.com/gh/missing-semicolon/ibis/642?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: ibis/bigquery/tests/test_compiler.py [hunk 2]
```diff
@@ -91,25 +90,25 @@ def test_to_timestamp(alltypes, timezone):
...
+                "TIMESTAMP '{}'".format('2017-01-01 04:55:59'),
+                dt.timestamp,
         ),
         (
-            '2017-01-01 04:55:59',
-            "TIMESTAMP '{}'".format('2017-01-01 04:55:59'),
-            dt.timestamp,
+                '2017-01-01 04:55:59',
+                "TIMESTAMP '{}'".format('2017-01-01 04:55:59'),
+                dt.timestamp,
```
Comment (cpcloud) [inline ibis/bigquery/tests/test_compiler.py]: Revert this change

File: ibis/bigquery/tests/test_compiler.py [hunk 3]
```diff
@@ -119,28 +118,87 @@ def test_literal_date(case, expected, dtype):
...
+                datetime.datetime(2017, 1, 1, 4, 55, 59),
+                "TIMESTAMP '{}'".format('2017-01-01 04:55:59'),
+                dt.timestamp,
+                'FORMAT_TIMESTAMP'
+        ),
+        (
+                '2017-01-01 04:55:59',
+                "TIMESTAMP '{}'".format('2017-01-01 04:55:59'),
+                dt.timestamp,
+                'FORMAT_TIMESTAMP'
```
Comment (cpcloud) [inline ibis/bigquery/tests/test_compiler.py]: Revert these spacing changes.

File: ibis/bigquery/tests/test_compiler.py [hunk 4]
```diff
@@ -119,28 +118,87 @@ def test_literal_date(case, expected, dtype):
...
+                "TIMESTAMP '{}'".format('2017-01-01 04:55:59'),
+                dt.timestamp,
         ),
         (
-            datetime.time(4, 55, 59),
-            "TIME '{}'".format('04:55:59'),
-            dt.time,
+                datetime.time(4, 55, 59),
+                "TIME '{}'".format('04:55:59'),
+                dt.time,
```
Comment (cpcloud) [inline ibis/bigquery/tests/test_compiler.py]: Revert these spacing changes

File: ibis/bigquery/compiler.py [hunk 5]
```diff
@@ -452,6 +449,19 @@ def _trans_param(self, expr):
 rewrites = BigQueryExprTranslator.rewrites
 
 
+@compiles(ops.DayOfWeekIndex)
+def bigquery_day_of_week_index(t, e):
+    arg = e.op().args[0]
+    arg_formatted = t.translate(arg)
+    return 'MOD(EXTRACT(DAYOFWEEK from {}) + 5, 7)'.format(arg_formatted)
```
Comment (cpcloud) [inline ibis/bigquery/compiler.py]: capitalize the `from` here

File: ibis/bigquery/compiler.py [hunk 6]
```diff
@@ -334,6 +331,7 @@ def truncator(translator, expr):
             )
 
         return "{}_TRUNC({}, {})".format(kind, arg, unit)
+
```
Comment (cpcloud) [inline ibis/bigquery/compiler.py]: Generally speaking, remove the whitespace changes. There was only a single whitespace lint error in CI and it was around the new `@compiles`/`@rewrites` functions that you introduced

**Comment by cpcloud** [comment]
@missing-semicolon can you resolve the conflicts here? i'll merge after you do that

**Comment by cpcloud** [comment]
@missing-semicolon Thanks! Merging.
